spikeinterface/core/frameslicesorting.py

```diff
@@ -11,27 +12,67 @@ class FrameSliceSorting(BaseSorting):
+    a corresponding sliced recording is registered to the sliced sorting.
+
+    Note that the returned sliced sorting may be empty.
+
+    Parameters
+    ----------
+    parent_sorting: BaseSorting
+    start_frame: None or int
+        Earliest included frame in the parent sorting(/recording).
+        Spike times(/traces) are re-referenced to start_frame in the
+        sliced objects. Set to 0 by default.
+    end_frame: None or int
+        Latest frame in the parent sorting(/recording). As for usual
+        python slicing, the end frame is excluded (such that the max
+        spike frame in the sliced sorting is `end_frame - start_frame - 1`)
+        If None (default), the end_frame is either:
+            - The total number of samples, if a recording is assigned
+            - The maximum spike frame + 1, if no recording is assigned
     """
 
     def __init__(self, parent_sorting, start_frame=None, end_frame=None):
         unit_ids = parent_sorting.get_unit_ids()
 
         assert parent_sorting.get_num_segments() == 1, 'FrameSliceSorting work only with one segment'
 
-        if start_frame is not None or end_frame is None:
-            parent_size = 0
-            for u in parent_sorting.get_unit_ids():
-                parent_size = np.max([parent_size, np.max(parent_sorting.get_unit_spike_train(u))])
 
         if start_frame is None:
             start_frame = 0
-        else:
-            assert 0 <= start_frame < parent_size
+        assert 0 <= start_frame, "Invalid value for start_frame: expected positive integer."
 
-        if end_frame is None:
-            end_frame = parent_size + 1
+        if parent_sorting.has_recording():
+            # Pull df end_frame from recording
+            parent_n_samples = parent_sorting._recording.get_total_samples()
+            if end_frame is None:
+                end_frame = parent_n_samples
+            assert end_frame <= parent_n_samples, (
+                "`end_frame` should be smaller than the sortings total number of samples."
+            )
+            assert start_frame <= parent_n_samples, (
+                "`start_frame` should be smaller than the sortings total number of samples."
+            )
+            if has_exceeding_spikes(parent_sorting._recording, parent_sorting):
+                raise ValueError(
+                    "The sorting object has spikes exceeding the recording duration. You have to remove those spikes "
+                    "with the `spikeinterface.curation.remove_excess_spikes()` function"
+                )
         else:
-            assert end_frame > start_frame, "'start_frame' must be smaller than 'end_frame'!"
+            # Pull df end_frame from spikes
+            if end_frame is None:
+                max_spike_time = 0
+                for u in parent_sorting.get_unit_ids():
+                    max_spike_time = np.max([max_spike_time, np.max(parent_sorting.get_unit_spike_train(u))])
+                end_frame = max_spike_time + 1
```

Comment:
This has a high cost at every  init which is not so godd for paralel processing.
If we have 500units and n_jobs=80 then we will make 40k np.max at every multiprocessing!!!
We should find a smarter way.
Maybe defer this computation at the last moement (get_traces).
@alejoe91 @TomBugnon what do you think ?

Reply:
Oups. No forget it.
The en_frame is store in the kwargs